packages/bitcore-wallet-service/src/lib/server.ts

```diff
@@ -3307,11 +3305,14 @@ export class WalletService implements IWalletService {
     } else {
       this.storage.fetchPendingTxs(this.walletId, (err, txps) => {
         if (err) return cb(err);
-
+        if (opts.tokenAddress) {
+          txps = txps.filter(txp => opts.tokenAddress === txp.tokenAddress);
+        } else {
+          txps = txps.filter(txp => !txp.tokenAddress);
```

Comment:
This breaks the nonce check in server.ts line 3059. I think we can just remove this line and only filter if opts.tokenAddress was given.